src/foam/nanos/script/Script.js

```diff
@@ -227,7 +227,7 @@ foam.CLASS({
 
           with ( { log: log, print: log, x: self.__context__ } ) {
             this.status = this.ScriptStatus.RUNNING;
-            var ret = eval(this.code);
+            this.runScript(x);
```

Comment:
If you are calling ``this.runScript`` here you need to have a js ``runScript`` function in Script.js.